libraries/wallet/wallet.cpp

```diff
@@ -1142,7 +1144,7 @@ class wallet_api_impl
 
       auto dyn_props = get_dynamic_global_properties();
       tx.set_reference_block( dyn_props.head_block_id );
-      tx.set_expiration( dyn_props.time + fc::seconds(30) );
+      tx.set_expiration( dyn_props.time + fc::seconds(transaction_expiration_time_sec) );
```

Comment:
I think we should refactor this code block by simply calling `sign_transaction()`.